src/lib/serviceRoutines/statisticsTreat.cpp

```diff
@@ -245,14 +247,17 @@ std::string renderSemWaitStats(void)
 std::string renderNotifQueueStats(void)
 {
   JsonHelper jh;
-
-  jh.addNumber("in",          QueueStatistics::getIn());
-  jh.addNumber("out",         QueueStatistics::getOut());
-  jh.addNumber("reject",      QueueStatistics::getReject());
-  jh.addNumber("sentOk",      QueueStatistics::getSentOK());     // FIXME P7: this needs to be generalized for all notificationModes
-  jh.addNumber("sentError",   QueueStatistics::getSentError());  // FIXME P7: this needs to be generalized for all notificationModes
-  jh.addFloat ("timeInQueue", QueueStatistics::getTimeInQ());
-  jh.addNumber("size",        QueueStatistics::getQSize());
+  float      timeInQ = QueueStatistics::getTimeInQ();
+  int        out     = QueueStatistics::getOut();
+
+  jh.addNumber("in",             QueueStatistics::getIn());
+  jh.addNumber("out",            out);
+  jh.addNumber("reject",         QueueStatistics::getReject());
+  jh.addNumber("sentOk",         QueueStatistics::getSentOK());     // FIXME P7: this needs to be generalized for all notificationModes
+  jh.addNumber("sentError",      QueueStatistics::getSentError());  // FIXME P7: this needs to be generalized for all notificationModes
+  jh.addFloat ("timeInQueue",    timeInQ);
+  jh.addFloat ("avgTimeInQueue", out==0 ? 0 : (timeInQ/out));
```

Comment:
Is the timeInQ/out casting correctly to float? I have found problems with this kind of things sometimes with C/C++.

Just to check that you have actually test it and it works ;)